### PR TITLE
Added test to check for alerts that are not enabled or does not have a notification

### PR DESCRIPTION
**Pull request by andrewalumkal**

### CODE REVIEW

**Comment by taddison** [comment]
Thanks @andrewalumkal - could you:
- Add something to the docs that gives a brief description of the test
- Provide some details on how you think the exclusion works.  I'm wondering if you want to list the alerts you care about vs. add alerts you want to ignore.  Scenario I'm thinking about is testing 'at least 1 alert for ID XYZ exists', rather than has someone an alert.

File: src/SQLChecks/Functions/Public/Get-UnconfiguredAlerts.ps1 [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+WHERE [enabled] = 0 OR [has_notification] = 0;
+"@
+
+    (Invoke-Sqlcmd -ServerInstance $ServerInstance -Query $query) | Where-Object {
+        $ExcludeAlert -notcontains $_.name
+      } | ForEach-Object {
+		[pscustomobject]@{
+			Name      = $_.name
+            Enabled   = $_.enabled
+            Has_Notification = $_.has_notification
```
Comment (taddison) [inline src/SQLChecks/Functions/Public/Get-UnconfiguredAlerts.ps1]: Could this be renamed


```suggestion
            HasNotification = $_.has_notification
```

File: src/SQLChecks/Tests/SQLInstance.tests.ps1 [hunk 2]
```diff
@@ -49,4 +49,10 @@ Describe "Database Mail is configured" -Tag DatabaseMail {
 	It "Database mail is running on $serverInstance" {
 		@(Get-DatabaseMailStatus -Config $Config).Status | Should Be "STARTED"
 	}
+}
+
+Describe "SQL Alerts" -Tag SQLAlerts {
```
Comment (taddison) [inline src/SQLChecks/Tests/SQLInstance.tests.ps1]: Could we pick a more descriptive tag - tags are globally unique in the module and drive config rules, so something that is verbose is better - maybe UnconfiguredSQLAgentAlerts ?

Comment (taddison) [inline src/SQLChecks/Tests/SQLInstance.tests.ps1]: Could we make this more descriptive - as it's unique on the module and frequently in config files with little context the more verbose the better.  UnconfiguredSQLAgentAlerts maybe?

**Review by taddison — CHANGES_REQUESTED** [review]

File: src/SQLChecks/Functions/Public/Get-UnconfiguredAlerts.ps1 (hunk 1 above)
Comment (andrewalumkal) [reply src/SQLChecks/Functions/Public/Get-UnconfiguredAlerts.ps1]: Updated

**Comment by andrewalumkal** [comment]
> Thanks @andrewalumkal - could you:
> 
> * Add something to the docs that gives a brief description of the test
> * Provide some details on how you think the exclusion works.  I'm wondering if you want to list the alerts you care about vs. add alerts you want to ignore.  Scenario I'm thinking about is testing 'at least 1 alert for ID XYZ exists', rather than has someone an alert.

Tim, so the check actually checks for ANY alert that is either disabled OR does not have an alert notification. The reason i picked this criteria is because if this was run against a new environment or after a migration, its good to know as an FYI what alerts are disabled OR which don't have a notification. 
The exclusion is to optionally exclude an alert we don't care about (report3 has log shipping alerts). The cleanest way to exclude would be to just delete the alert from SQL Server. But the exclusion is there in case you don't want to delete the alert, but keep it disabled / without a notification for some reason. Let me know if you would like to change the logic of this alert.

File: src/SQLChecks/Tests/SQLAgent.tests.ps1 [hunk 3]
```diff
@@ -20,4 +20,10 @@ Describe "SQL Agent status" -Tag AgentIsRunning {
     It "SQL Agent is running on $serverInstance" {
         (Get-SqlAgentService -Config $Config).Status | Should Be "Running"
     }
+}
+
+Describe "Unconfigured SQL Agent Alerts" -Tag UnconfiguredSQLAgentAlerts {
```
Comment (taddison) [inline src/SQLChecks/Tests/SQLAgent.tests.ps1]: This tag should match the documentation and function - `CheckUnconfiguredSQLAgentAlerts`

**Review by taddison — CHANGES_REQUESTED** [review]

**Review by taddison — APPROVED** [review]

**Comment by taddison** [comment]
Merged

**Pull request merged**